### PR TITLE
refactor: restructure rm package

### PR DESCRIPTION
**Pull request by stoksc**
## Description
Move RMs into their own packages. This change is mostly organizational, but it does help a lot in EE to understand the plugin nature of the dispatcher resource manager, and which files are shares and which are not.

## Test Plan
- [x] all existing tests should pass, pure refactor (i did refactor one test to get around circular imports, but did not change what it tested)

## Commentary (optional)
finally did this..

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/experiment.go [hunk 1]
```diff
@@ -804,7 +806,7 @@ func (e *experiment) setPriority(ctx *actor.Context, priority *int, forward bool
 			Handler:  ctx.Self(),
 		}).(type) {
 		case nil:
-		case rm.ErrUnsupported:
+		case rmerrors.ErrUnsupported:
```
Comment (stoksc) [inline master/internal/experiment.go:809]: open to better naming for any of the packages, didn't put an excessive amount of thought into it.

File: master/internal/rm/agentrm/agent_state.go [hunk 2]
```diff
@@ -718,7 +730,9 @@ func loadContainersToAllocationIds(
 	}
 
 	for _, row := range result {
-		rr[cproto.ID(row["container_id"].(string))] = model.AllocationID(row["allocation_id"].(string))
+		rr[cproto.ID(row["container_id"].(string))] = model.AllocationID(
```
Comment (stoksc) [inline master/internal/rm/agentrm/agent_state.go:733]: i just did a run of `golines` to fix all ll lints, it did some of this

**Review by ioga — APPROVED** [review]
oh wow `kubernetesrm` doesn't import `agentrm` anywhere?

**Comment by stoksc** [comment]
this could also use a what's public/what's private refactor.. maybe that is next up to bat

**Pull request merged**